src/components/CustomChannelComponent.tsx

```diff
@@ -227,72 +225,49 @@ export function CustomChannelComponent() {
-            welcomeMessages.length > 0
-          ) {
-            if (!firstMessageId || message.messageId === firstMessageId) {
-              const lastWelcomeMessageIndex = welcomeMessages.length - 1;
-              return (
-                <Message {...props} message={message}>
-                  {welcomeMessages.map((welcomeMsg, index) => {
-                    const suggestedReplies = welcomeMsg.suggestedReplies;
-                    // TODO: support file message in the future.
-                    if ('message' in welcomeMsg) {
-                      const text = welcomeMsg.message;
-                      const tokens: Token[] = parseTextMessage(
-                        text,
-                        replacementTextList
-                      );
-                      return (
-                        <div ref={lastMessageRef} key={index}>
-                          <BotMessageWithBodyInput
-                            chainTop={index === 0}
-                            chainBottom={index === lastWelcomeMessageIndex}
-                            messageCount={messageCount}
-                            botUser={botUser}
-                            bodyComponent={
-                              <ParsedBotMessageBody
-                                text={text}
-                                tokens={tokens}
-                              />
-                            }
-                            createdAt={message.createdAt}
-                          />
-                          {suggestedReplies && suggestedReplies.length && (
-                            <DynamicRepliesPanel
-                              replyOptions={suggestedReplies}
-                            />
-                          )}
-                        </div>
-                      );
-                    }
-                  })}
-                </Message>
-              );
-            }
-            return <></>;
-          }
           /**
            * Note that we display injected welcome messages regardless of first welcome message
            * is filtered.
            */
           // NOTE: Filter out messages that should not be displayed.
+          if (
+            isWelcomeMessagesGiven &&
+            botWelcomeMessageIds.includes(message.messageId)
+          )
+            return <></>;
           if (shouldFilterOutMessage(message)) return <></>;
           const groupedMessage = groupedMessages.find(
             (m) => m.messageId == message.messageId
           );
 
+          /* eslint-disable react/prop-types */ // TODO: upgrade to latest eslint tooling or add "react/prop-types": "off" to eslintrc rules.
+          let hasSeparator = props.hasSeparator;
+          const prevMessageTimestamp =
+            lastWelcomeMessageCreatedAt ?? firstMessageCreatedAt;
+          if (isWelcomeMessagesGiven && prevMessageTimestamp) {
+            hasSeparator = !isSameDay(message.createdAt, prevMessageTimestamp);
+          }
```

Comment:
hasSeparator logic is for deciding whether to add data separator or not (above message)